torchtune/modules/transforms/vision_utils/get_resize_dimensions.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+
+import math
+
+from typing import Optional, Tuple
+
+logger = logging.getLogger(__name__)
+
+
+def get_resize_dimensions(
+    image_size: Tuple[int], target_size: Tuple[int], max_upscaling_size: Optional[int]
+) -> Tuple[int]:
+    """
+    Calculates the resize dimensions for an image, given the target size and max_upscaling_size.
+    The user can set max_upscaling_size to limit upscaling when target_size exceeds image_size.
+
+    Args:
+        image_size (Tuple[int]): The size of the image, in the form [height, width].
+        target_size (Tuple[int]): The desired resolution to fit the image into, in the format [height, width].
+        max_upscaling_size (Optional[int]): The maximum size to upscale the image to.
+            If None, will upscale to target size.
+
+    Returns:
+        Tuple[int]: The resize dimensions for the image, in the format [height, width].
+
+    Examples:
+
+        Example 1: The image will be upscaled from (300, 800) to (448, 1194), since 448 is the limiting side.
+
+            >>> max_upscaling_size = None
+            >>> image_size = (300, 800)
+            >>> target_size = (448, 1344)
+            >>> output = get_resize_dimensions(image_size, target_size, resample, max_upscaling_size)
+
+        Example 2: The image will stay as is, since 800 > 600.
```

Comment:
@felipemello1 I find it counterintuitive that this doesn't cause a downsize to (225, 600). max_upscaling_size is a bit weird to have here at all because you could just not provide a target that's too big, but I understand it's convenient.